google-cloud-pubsublite/src/main/java/com/google/cloud/pubsublite/cloudpubsub/SubscriberSettings.java

```diff
@@ -113,18 +125,36 @@ Subscriber instantiate() throws StatusException {
     wireCommitterBuilder.setSubscriptionPath(subscriptionPath());
     cursorServiceStub().ifPresent(wireCommitterBuilder::setCursorStub);
 
-    List<Subscriber> perPartitionSubscribers = new ArrayList<>();
-    for (Partition partition : partitions()) {
-      wireSubscriberBuilder.setPartition(partition);
-      wireCommitterBuilder.setPartition(partition);
-      perPartitionSubscribers.add(
-          new SinglePartitionSubscriber(
+    PartitionSubscriberFactory partitionSubscriberFactory =
+        partition -> {
+          wireSubscriberBuilder.setPartition(partition);
+          wireCommitterBuilder.setPartition(partition);
+          return new SinglePartitionSubscriber(
               receiver(),
               transformer().orElse(MessageTransforms.toCpsSubscribeTransformer()),
               new AckSetTrackerImpl(wireCommitterBuilder.build()),
               nackHandler().orElse(new NackHandler() {}),
               messageConsumer -> wireSubscriberBuilder.setMessageConsumer(messageConsumer).build(),
-              perPartitionFlowControlSettings()));
+              perPartitionFlowControlSettings());
+        };
+
+    if (!partitions().isPresent()) {
+      AssignerBuilder.Builder assignerBuilder = AssignerBuilder.newBuilder();
+      assignerBuilder.setSubscriptionPath(subscriptionPath());
+      assignmentServiceStub().ifPresent(assignerBuilder::setAssignmentStub);
+      AssignerFactory assignerFactory =
+          receiver -> {
+            assignerBuilder.setReceiver(receiver);
+            return assignerBuilder.build();
+          };
+      return new AssigningSubscriber(partitionSubscriberFactory, assignerFactory);
+    }
+
+    List<Subscriber> perPartitionSubscribers = new ArrayList<>();
+    for (Partition partition : partitions().get()) {
+      wireSubscriberBuilder.setPartition(partition);
+      wireCommitterBuilder.setPartition(partition);
```

Comment:
I don't think we need these lines anymore:

```
wireSubscriberBuilder.setPartition(partition);
wireCommitterBuilder.setPartition(partition);
```

This should be covered by the partitionSubscriberFactory, right?

Reply:
Nack. This is still needed in the case when the user explicitly provides a list of partitions.

Reply:
Agreed we still need to set the partition, but we already do this inside partitionSubscriberFactory [L130-131].  When we pass the partition to the partitionSubscriberFactory below, we will again set the partition on the wireSubscriberBuilder and wireCommitterBuilder.  Or am I missing something?